forum/startups/views.py

```diff
@@ -1,13 +1,81 @@
 from rest_framework import generics, status
+from rest_framework.views import APIView
 from rest_framework.response import Response
 from rest_framework.permissions import IsAuthenticated
 from .models import Startup
+from investors.models import Investor, InvestorFollow
 from .serializers import StartupSerializer
 import logging
 
 logger = logging.getLogger(__name__)
 
 
+class SavedStartupsListAPIView(APIView):
+    """
+    API endpoint for listing all startups an investor has saved.
+
+    Permissions:
+        - Requires the user to be authenticated.
+
+    Methods:
+        - GET: Retrieves and returns a list of startups that the authenticated investor has saved.
+        
+    Workflow:
+        1. The view first retrieves the `Investor` instance associated with the authenticated user.
+        2. It then filters startups linked to the investor in the `InvestorFollow` model.
+        3. The filtered startups are serialized and returned as a JSON response.
+        
+    Responses:
+        - 200 OK: Returns a list of saved startups if the investor is found.
+        - 404 Not Found: Returns an error message if the investor does not exist.
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        try:
+            investor = Investor.objects.get(user=request.user)
+            saved_startups = Startup.objects.filter(startup_investors__investor=investor)
+            serializer = StartupSerializer(saved_startups, many=True)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        except Investor.DoesNotExist:
+            return Response({"error": "Investor not found."}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
The `SavedStartupsListAPIView` could cause performance issues if the investor has many saved startups. Adding pagination would improve user experience and performance.